plugins/aiops/pkg/gateway/admin.go

```diff
@@ -338,14 +341,38 @@ func (s *AIOpsPlugin) DeleteAISettings(ctx context.Context, _ *emptypb.Empty) (*
 		},
 	})
 	if client.IgnoreNotFound(err) != nil {
-		return nil, err
+		return err
 	}
 
 	err = s.k8sClient.DeleteAllOf(ctx, &aiv1beta1.PretrainedModel{}, client.InNamespace(s.storageNamespace))
 	if err != nil {
-		return nil, err
+		return err
+	}
+	return nil
+}
+
+func (s *AIOpsPlugin) DeleteAISettings(ctx context.Context, options *admin.DeleteOptions) (*emptypb.Empty, error) {
+	var deleteEg util.MultiErrGroup
+	if lo.FromPtrOr(options.PurgeModelTrainingData, false) {
+		deleteEg.Go(func() error {
+			ctxca, ca := context.WithTimeout(ctx, 10*time.Second)
+			defer ca()
+			if err := s.deleteTrainingJobInfo(ctxca); err != nil {
+				s.Logger.Error(fmt.Sprintf("failed to purge %s", err))
+				return err
+			}
+			return nil
+		})
 	}
-	return &emptypb.Empty{}, nil
+	deleteEg.Go(func() error {
+		if err := s.deleteAIOpsResources(ctx); err != nil {
+			s.Logger.Error(fmt.Sprintf("failed to delete AiOps resources : %s", err))
+			return err
+		}
+		return nil
+	})
+	deleteEg.Wait()
+	return &emptypb.Empty{}, deleteEg.Error()
```

Comment:
For the UI flow this should be serialized with the KV delete occurring first (as discussed)